playbooks/phobos_vpn.yml

```diff
@@ -1,35 +1,95 @@
 - name: Configure Phobos VPN Tunnel (should only be run on single use slaves)
   hosts: "{{ target_hosts | default('localhost') }}"
   tasks:
-    # Use shell because python-apt isn't available so the apt module fails.
-    # I don't want to add python-apt to one of the more general slave setup
-    # scripts as it depends on python. We want to reduce the number of deps
-    # installed on general slaves to the bare minimum.
-    - name: Install apt packages
+    - name: Install netcat
       shell: |
         apt-get update
-        apt-get install -y vpnc
-      register: install_packages
-      until: install_packages | success
+        apt-get install -y netcat-openbsd
```

Comment:
Rather than install netcat, you could just use wait_for to test connectivity. As noted in the comment that's been removed here, we try not to install things on target hosts as far as we possibly can.

Reply:
1. the comment wasn't removed, but instead moved along with the associated task (and placed within a `block:`, ie indented, which misled the diff algorithm, which is why i recommended diff-ing with the "hide whitespace changes" option, whether here in github or in git with `-w`).
1. i interpreted the comment to refer to packages installed on the slave *before* a job (as part of slave setup), not *during* a job, especially if easily worked around, not needed for all jobs, and increases "base" slave resources (eg setup time, image size if making snapshots or creating custom images).
1. `wait_for` doesn't appear to support udp which is what vpnc uses (ie udp port 4500).

all that said, it was my mistake to use netcat for udp port detection (`UDP port scans using the -uz combination of flags will always report success irrespective of the target machine's state.` - netcat man page; i thought its algorithm was better, more like nmap's where it factors in ICMP responses, eg "destination port unreachable" if the port is closed, but firewalls and IP stack configurations make this unreliable), so i'm going back to:

1. install ping
1. use ping to detect VPN server reachability

Reply:
Ah, fair enough. Thanks for covering the bases - I think using ping might be the simplest option here then.